src/CodeGen_Metal_Dev.cpp

```diff
@@ -17,6 +18,16 @@ using std::sort;
 using std::string;
 using std::vector;
 
+// Storage for Metal compilation tools
+namespace {
+struct MetalTools {
+    std::string compiler;
+    std::string linker;
+};
+}  // namespace
+
+MetalTools metal_tools;
```

Comment:
I'm confused by this pattern. Why make a type private to this file but not the only instance of it?